### ISSUE TITLE
Not using the right python environment in projects

### ISSUE THREAD

**Issue by pranasziaukas** [python-poetry/poetry#4918]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS v12.1
- **Poetry version**: v1.1.12
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/pranasziaukas/1aa612116b00d49ff0cf7ac0511cd040

## Issue

After a fresh install on macOS
```
➜ curl -sSL https://install.python-poetry.org | python3 -
```
`poetry` was added to `~/Library/Python/3.9/bin`, which I thought was odd, but no complaints as long as it works:
```
➜ poetry --version
Poetry version 1.1.12
```

Created a default project `pyproject.toml` via `poetry init`, just changed the python version constraint to
```
python = "^3.10,<3.11"
```

I leverage `pyenv` to switch between different versions of python, so I did just that:
```
➜ pyenv local 3.10.0
➜ python --version
Python 3.10.0
```

At this point, I'd expect `poetry` to rely on the locally activated python version. But it doesn't:
```
➜ poetry install -vvv
The currently activated Python version 3.9.9 is not supported by the project (^3.10,<3.11).
Trying to find and use a compatible version.
Trying python3
Using python3 (3.10.0)
Creating virtualenv test-5f6I1Ta7-py3.10 in /Users/pranas/Library/Caches/pypoetry/virtualenvs
Using virtualenv: /Users/pranas/Library/Caches/pypoetry/virtualenvs/test-5f6I1Ta7-py3.10
Updating dependencies
Resolving dependencies...
   1: fact: test is 0.1.0
   1: derived: test
   1: selecting test (0.1.0)
   1: Version solving took 0.002 seconds.
   1: Tried 1 solutions.

Writing lock file

Finding the necessary packages for the current system
```

It claims that python v3.9.9 is activated, which is not true.
And the environment is wrong anyway:
```
➜ poetry env info

Virtualenv
Python:         3.9.9
Implementation: CPython
Path:           NA

System
Platform: darwin
OS:       posix
Python:   /usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9
```

So, effectively, the appropriate python 3.10.0 is not accessible at all:
```
➜ poetry run python --version
The currently activated Python version 3.9.9 is not supported by the project (^3.10,<3.11).
Trying to find and use a compatible version.
Using python3 (3.10.0)
The virtual environment found in /usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9 seems to be broken.
Recreating virtualenv test-5f6I1Ta7-py3.10 in /Users/pranas/Library/Caches/pypoetry/virtualenvs/test-5f6I1Ta7-py3.10
Python 3.9.9
```

**Comment by finswimmer** [comment]
TL;DR: The "pyenv" no longer works with the `install-poetry.py` script. See:

* https://github.com/python-poetry/poetry/issues/4199
* https://github.com/python-poetry/poetry/issues/651#issuecomment-864533910
* https://github.com/python-poetry/poetry/issues/4187#issuecomment-863881765

We are discussing possible solutions:

* https://github.com/python-poetry/poetry/pull/4179
* https://github.com/python-poetry/poetry/pull/4852
